pkgs/jnigen/lib/src/elements/simple_elements.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'elements.dart';
+
+// as the list of methods and the list of fields is not modifiable and not final
+extension on ClassDecl {
+  void updateMethods(List<Method> methods) {
+    this.methods = methods;
+  }
+
+  void updateFields(List<Field> fields) {
+    this.fields = fields;
+  }
+}
+
+abstract class Element {
+  void accept(Visitor visitor);
+}
+
+// making the functions empty, as if the user will not use some of them
+// he is not required to override them
+abstract class Visitor {
+  void visitClass(SimpleClassDecl c) {}
+  void visitMethod(SimpleMethod method) {}
+  void visitField(SimpleField field) {}
+}
+
+class SimpleClasses implements Element {
+  SimpleClasses(this._classes);
+  final Classes _classes;
+
+  @override
+  void accept(Visitor visitor) {
+    final excludedClassesKey = <String>[];
+    for (var entry in _classes.decls.entries) {
+      final c = entry.value;
+      final simpleClassDecl = SimpleClassDecl(c);
+      simpleClassDecl.accept(visitor);
+      if (simpleClassDecl.isExcluded) {
+        excludedClassesKey.add(entry.key);
+      }
+    }
+    for (var key in excludedClassesKey) {
+      _classes.decls.remove(key);
```

Comment:
I'd say removing them entirely from the original ast might not be the best choice. Maybe another visitor later on wants to include them again. Also if someone excludes an outer class, we need to exclude inner classes as well.

Overall, maybe it's best to set a flag in the original AST node which then will be respected by the `excluder.dart`.